src/simulate.jl

```diff
@@ -69,18 +91,80 @@ function propagate!(sim::NestedFilterSimulation, dt)
 end
 
 function run!(sim::NestedFilterSimulation; T::Int, plot_each_timestep::Bool = false)
+    entropies = Entropies(zeros(T),zeros(T),zeros(T),zeros(T),zeros(T))
+    Maps = Map(zeros(T),zeros(T),zeros(T),zeros(T),zeros(T))
+            
+    results = Results(entropies,Maps,zeros(T))
+    
     if length(sim.times) == 0
         initialize!(sim)
     end
     for i in 1:T
         begin
-            propagate!(sim)
+            time = @timed propagate!(sim)
         end
         if plot_each_timestep
             posterior_plot(sim)
         end
+        save_results!(results, sim, i, time)
+        if i == T
+            save(string(Base.parse(Int, ENV["SLURM_ARRAY_TASK_ID"]),".jld"), "entropies", results.entropies, "ISI", sim.times, "MAP", results.map)
+        end
     end
     return sim.times, sim.epsps
 end
 
+function save_results!(results::Results, sim::NestedFilterSimulation, i, time)
+    
+    results.time[i] = time
+    
+    Nind = Array(sim.fstate.model.Nind)
+    Nrng = Array(sim.fstate.model.Nrng)
+    N_posterior = zeros(length(Nrng))
+    for j in 1:length(Nrng)
+        N_posterior[j] = count(i->(i==j),Nind)
+    end
+    results.entropies.entropy_N[i] = entropy(N_posterior/sum(N_posterior))  
+
+    pind = Array(sim.fstate.model.pind)
+    prng = Array(sim.fstate.model.prng)
+    p_posterior = zeros(length(prng))
+    for j in 1:length(prng)
+        p_posterior[j] = count(i->(i==j),pind)
+    end
+    results.entropies.entropy_p[i] = entropy(p_posterior/sum(p_posterior))  
+
+    qind = Array(sim.fstate.model.qind)
+    qrng = Array(sim.fstate.model.qrng)
+    q_posterior = zeros(length(qrng))
+    for j in 1:length(qrng)
+        q_posterior[j] = count(i->(i==j),qind)
+    end
+    results.entropies.entropy_q[i] = entropy(q_posterior/sum(q_posterior))  
+
+    σind = Array(sim.fstate.model.σind)
+    σrng = Array(sim.fstate.model.σrng)
+    σ_posterior = zeros(length(σrng))
+    for j in 1:length(σrng)
+        σ_posterior[j] = count(i->(i==j),σind)
+    end
+    results.entropies.entropy_σ[i] = entropy(σ_posterior/sum(σ_posterior))  
+  
+    τind = Array(sim.fstate.model.τind)
+    τrng = Array(sim.fstate.model.τrng)
+    τ_posterior = zeros(length(τrng))
+    for j in 1:length(τrng)
+        τ_posterior[j] = count(i->(i==j),τind)
+    end
+    results.entropies.entropy_τ[i] = entropy(τ_posterior/sum(τ_posterior))    
+
+    map = MAP(sim.fstate.model)
+    results.map.map_N[i] = map.N
+    results.map.map_p[i] = map.p
+    results.map.map_q[i] = map.q
+    results.map.map_σ[i] = map.σ
+    results.map.map_τ[i] = map.τ
+
```

Comment:
Add `return` statement, otherwise the function will return the last result, i.e. `map.τ` in this case. I would suggest `return results`.

Reply:
Added return results statement